src/background.jl

```diff
@@ -90,88 +90,90 @@ over the redshift range specified by the `CosmologicalGrid`.
 # Notes:
 This function modifies the `BackgroundQuantities` struct in place by filling the arrays with the computed values.
 """
-function evaluate_background_quantities!(CosmologicalGrid::CosmologicalGrid,
-    BackgroundQuantities::BackgroundQuantities,
-    AbstractCosmology::AbstractCosmology)
-    for z_idx in 1:length(CosmologicalGrid.z_range)
+function evaluate_background_quantities!(grid::CosmologicalGrid,
+    bkgq::BackgroundQuantities, cosmo::AbstractCosmology)
+    for z_idx in 1:length(grid.z_range)
         # Compute the Hubble parameter H(z)
-        BackgroundQuantities.Hz_array[z_idx] = compute_hubble_factor(
-            CosmologicalGrid.z_range[z_idx], AbstractCosmology)
+        bkgq.Hz_array[z_idx] = compute_hubble_factor(grid.z_range[z_idx], cosmo)
         
         # Compute the comoving distance χ(z)
-        BackgroundQuantities.χz_array[z_idx] = compute_χ(
-            CosmologicalGrid.z_range[z_idx], AbstractCosmology)
+        bkgq.χz_array[z_idx] = compute_χ(grid.z_range[z_idx], cosmo)
```

Comment:
This part is fine now, but we are gonna have to modify later. First, you are not using vectorization. Second, you are mutating arrays, which does not work nicely with `Zygote.jl`.  Add a TODO.